src/go/rpk/pkg/plugin/manifest.go

```diff
@@ -140,3 +146,75 @@ func (p *ManifestPlugin) PathSha(
 
 	return fmt.Sprintf("/%s/%s/%s/%s", p.Path, need, sha, p.Name), sha, nil
 }
+
+// DownloadForUser is a shortcut for Download for the current user's
+// os / arch.
+func (p *ManifestPlugin) DownloadForUser(baseURL string) ([]byte, error) {
+	return p.Download(baseURL, runtime.GOOS, runtime.GOARCH)
+}
+
+// Download downloads and decompresses the plugin represented by this manifest
+// entry.
+func (p *ManifestPlugin) Download(baseURL, os, host string) ([]byte, error) {
+	var decompress bool
+	switch p.Compression {
+	case "":
+		// do nothing
+	case "gzip":
+		decompress = true
+	default:
+		return nil, fmt.Errorf("plugin has compression %q, which we do now know how to decompress", p.Compression)
+	}
+
+	path, sha, err := p.PathSha(os, host)
+	if err != nil {
+		return nil, err // error already annotated
+	}
+
+	u := baseURL + path
+	req, err := http.NewRequestWithContext(
+		context.Background(),
+		http.MethodGet,
+		u,
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create request %s: %v", u, err)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("unable to issue request %s: %v", u, err)
+	}
+	if resp.StatusCode/100 != 2 {
+		return nil, fmt.Errorf("unsuccessful plugin response %s to request %s", http.StatusText(resp.StatusCode), u)
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read response for request %s: %v", u, err)
+	}
+
+	if decompress {
+		gzr, err := gzip.NewReader(bytes.NewBuffer(body))
+		if err != nil {
+			return nil, fmt.Errorf("unable to create gzip reader: %w", err)
+		}
+		if body, err = io.ReadAll(gzr); err != nil {
+			return nil, fmt.Errorf("unable to gzip decompress plugin: %w", err)
+		}
+		if err = gzr.Close(); err != nil {
+			return nil, fmt.Errorf("unable to close gzip reader: %w", err)
+		}
+	}
+
+	shasum := sha256.Sum256(body)
+	gotsha := strings.ToLower(hex.EncodeToString(shasum[:]))
+	expsha := strings.ToLower(sha)
+
+	if gotsha != expsha {
+		return nil, fmt.Errorf("download of %s: sha sum %s != manifest sha sum %s", u, gotsha, expsha)
```

Comment:
Let's change this for something more human-readable. How about
```suggestion
		return nil, fmt.Errorf("the checksum of the downloaded plugin doesn't match the expected one)
```

And maybe we can log the 2 checksums at debug level?

Reply:
The shasums are pretty valuable to print. Odds are a person will just copy this error message to us in a support ticket if it is persistent. I wouldn't want to rely on them opting in to debug logging.

I think a push with the phrasing
```go
fmt.Errorf("checksum of plugin %s does not match what the manifest specifies (downloaded sha256sum: %s, manifest specified sha256sum: %s)", u, gotsha, expsha)
```
is clearer.